mds/views.py

```diff
@@ -109,38 +110,63 @@ def perform_update(self, serializer):
         serializer.save()
 
 
+class DeviceLimitOffsetPagination(LimitOffsetPagination):
+    default_limit = 500
+
+
 class DeviceViewSet(
-    mixins.CreateModelMixin,
-    mixins.RetrieveModelMixin,
-    mixins.DestroyModelMixin,
-    mixins.ListModelMixin,
-    UpdateOnlyModelMixin,
-    MultiSerializerViewSet,
+    viewsets.ModelViewSet, UpdateOnlyModelMixin, MultiSerializerViewSet
 ):
-
-    queryset = models.Device.objects.prefetch_related(
-        Prefetch(
-            "telemetries",
-            queryset=models.Telemetry.objects.filter(
-                id__in=Subquery(
-                    models.Telemetry.objects.filter(device_id=OuterRef("device_id"))
-                    .order_by("-timestamp")
-                    .values_list("id", flat=True)[:1]
+    def get_queryset(self):
+        queryset = models.Device.objects
+
+        filters = {
+            "category": "category__in",
+            "status": "telemetries__status__in",
+            "provider": "provider__id__in",
+        }
+        time_filters = {
+            "registrationDateFrom": "registration_date__gte",
+            "registrationDateTo": "registration_date__lte",
+        }
```

Comment:
Why not use DRF style Filter classes the same way you did for pagination ? 
The number of filters needed here seem to point to a classic use case for a [Generic Filter](https://www.django-rest-framework.org/api-guide/filtering/#generic-filtering)

Reply:
Yes, django-filter does pretty much all the work I needed